modules/ROOT/pages/http/http-listener-ref.adoc

```diff
@@ -1,182 +1,296 @@
 ----
 
-The request headers are:
+The resulting HTTP request when submitting the form is a multipart one:
 
 [source,text,linenums]
 ----
-POST /accounts HTTP/1.1
+POST /api/v1/account/mulesoft/logo HTTP/1.1
+Content-Type: multipart/form-data; boundary=489691234097965980223899
 Host: localhost:8081
-Cache-Control: no-cache
-User-Agent: Mule/4.0
-Content-Type: multipart/form-data; boundary=AaB03x
+content-length: 34332
 
- --AaB03x
+--489691234097965980223899
 Content-Disposition: form-data; name="name"
 
-MuleSoft
- --AaB03x
-Content-Disposition: form-data; name="request"; filename="myrequest.png"
+Corporate Logo
+--489691234097965980223899
+Content-Disposition: form-data; name="logo"; filename="MuleSoft_logo.png"
 Content-Type: image/png
 
-
- --AaB03x
+.PNG
+.
+...
+IHDR.......L......~.....	pHYs...#...#.x.?v.. .IDATx....q.W.6.....~".N....t....t..#.....LD0T.CF0b..:.3......Q..@...q]U*y\c....
+....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%
+.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....V"H....`%.$....^6.......|..P.....IEND.B`.
+--489691234097965980223899
 Content-Disposition: form-data; name="color"
 
 blue
- --AaB03x
+--489691234097965980223899--
 ----
 
-== Using DataWeave to Access the Form Data
-
-You access the form values using expressions that resemble these:
+Each item can be accessed through the `parts` object, either by name or item number. For example, the second part can be accessed with `payload.parts.logo` or `payload.parts[1]`. The latter is useful when a name is not provided.
+Within each part, you can access its content and headers. For example, `payload.parts.color.content` will return `blue` while `payload.parts.logo.headers.'Content-Type'` will return `application/png`.
+A very common scenario is obtaining the filename of a part, for this reason, the `Content-Disposition` header is parsed to allow an expression like `payload.parts.logo.headers.'Content-Disposition'.filename` which in this case will return `MuleSoft_logo.png`.
 
-* Access the content of the `name` field
-+
-`#[payload.name]`
-+
-* Access the headers
-+
-`#[attributes.headers['content-type']`
+See the xref:4.1@mule-runtime/dataweave-formats.adoc[Formats Supported by DataWeave] to learn more about reading and writing multipart content.
 
-For more information, see "About HTTP Responses to Requests."
+=== About HTTP Responses
 
-== HTTP Custom Properties Request Line
+After the triggered flow is done processing, its result is returned so that the listener can provide a response. If the flow executed successfully, a regular response will be returned but if the flow terminated with a failure then an error response will be returned.
+By default, a regular response will return a 200 status code and the message payload as body, while an error response will return a 500 status code with the flow error's description as body. These responses can be customized by providing:
```

Comment:
a common ask is how to set the response code based on some condition. Would be good to add a simple example of that. (e.g: `statusCode="#[vars.recordInserted ? '201' : '200']"` , or however a ternary is written in DW)